zerver/lib/avatar.py

```diff
@@ -36,6 +34,33 @@ def avatar_url(
     )
 
 
+def get_system_bots_avatar_file_name(email: str) -> str:
+    system_bot_avatar_name_map = {
+        settings.WELCOME_BOT: "welcome-bot",
+        settings.NOTIFICATION_BOT: "notification-bot",
+        settings.EMAIL_GATEWAY_BOT: "emailgateway",
+    }
+    return urljoin(STATIC_AVATARS_DIR, system_bot_avatar_name_map.get(email, "unknown"))
+
+
+def get_static_avatar_url(email: str, medium: bool) -> str:
+    avatar_file_name = get_system_bots_avatar_file_name(email)
+
+    avatar_file_name += "-medium.png" if medium else ".png"
+
+    if not DEBUG and not staticfiles_storage.exists(avatar_file_name):
+        # This logic cannot be executed while STATIC_ROOT is
+        # not defined.
+        return DEFAULT_AVATAR_FILE
+    else:
+        from django.contrib.staticfiles.finders import find
+
+        if not find(avatar_file_name):
```

Comment:
I couldn't import `static_path` for this due to circular import.

Reply:
I don't think we want to call this `find` operation on every request for the relevant avatar; that function may be expensive. I think it would be reasonable to put this as `if DEBUG and not find(...)`? 

The other option would be to try to do it at most once per process, rather than once per request, but I think that'd be more complexity than it's worth.